### PR TITLE
Refactor tests structure, include tests in dist tarball. Fixes #212

### LINKED ISSUE #212: Improvements to test suite
Currently, the tests are starting to be a huge unreadable blob. I suggest:
- splitting `unit_test.py` into more modules (each module should cover one file from `osbs`)
- using test classes to group together tests for functional classes
- and maybe some more stuff; increasing coverage would also be nice :)

I'll try to work on this when I have some time, but if someone wants to take this sooner, please speak up.

### PR DESCRIPTION
**Pull request by bkabrda**
This PR splits the tests to multiple files/classes and also includes tests in sdist (they will be in tarball, but won't install with `pip install osbs-client`).

This way, tests are better structured. It's now more easy to run tests for only separate part of osbs, e.g.:
- `py.test -k test_http`
- `py.test -k TestHttpSession`

Also, it's much easier to see what is tested and what is not even without having to see coverage report.

### CODE REVIEW

File: tests/build/test_build_request.py [hunk 1]
```diff
@@ -0,0 +1,596 @@
+import copy
```
Comment (twaugh) [inline tests/build/test_build_request.py]: No copyright notice.

File: tests/constants.py [hunk 2]
```diff
@@ -7,6 +7,11 @@
 """
 from __future__ import absolute_import, unicode_literals
 
+import os
+
+HERE = os.path.dirname(__file__)
+INPUTS_PATH = os.path.join(HERE, '..', 'inputs')
```
Comment (twaugh) [inline tests/constants.py]: Great idea :-)

File: tests/test_http.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+from flexmock import flexmock
+import pycurl
+import pytest
+
+import osbs.http as osbs_http
+from osbs.http import parse_headers, HttpSession, HttpStream
+
+from tests.fake_api import Connection, ResponseMapping
+
+logger = logging.getLogger("osbs.tests")
```
Comment (twaugh) [inline tests/test_http.py]: Why not `logging.getLogger(__name__)` here?

**Comment by twaugh** [comment]
I've added some comments above.

File: tests/test_http.py (hunk 3 above)
Comment (bkabrda) [inline tests/test_http.py]: I don't know, I copy-pasted that from the previous version of tests :) I will fix that.

**Comment by bkabrda** [comment]
Thanks, I addressed the issues in the 2 subsequent commits.

**Comment by twaugh** [comment]
LGTM

**Pull request merged**